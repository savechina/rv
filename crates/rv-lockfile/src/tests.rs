#[test]
fn test_parse_file() {
    let input = include_str!("../tests/inputs/Gemfile.lock.test0");
    let output = crate::parse(input).unwrap();
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_file_two_sources() {
    let input = include_str!("../tests/inputs/Gemfile.lock.twosources");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_empty_sections() {
    let input = include_str!("../tests/inputs/Gemfile.lock.empty");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_with_checksums() {
    let input = include_str!("../tests/inputs/Gemfile.lock.withchecksums");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_with_paths() {
    let input = include_str!("../tests/inputs/Gemfile.lock.withpath");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_feedyouremail() {
    let input = include_str!("../tests/inputs/Gemfile.lock.feedyouremail");
    let output = must_parse(input);
    assert_eq!(output.dependencies.len(), 52);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_gitlab() {
    let input = include_str!("../tests/inputs/Gemfile.lock.gitlab");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_gemdir() {
    let input = include_str!("../tests/inputs/Gemfile.lock.gemdir");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_discourse() {
    let input = include_str!("../tests/inputs/Gemfile.lock.discourse");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

#[test]
fn test_parse_withoutsource() {
    // If the Gemfile has no declared source, Bundler will default to http://rubygems.org,
    // which provides the endpoints needed to resolve a lockfile successfully, but does not
    // provide the endpoints needed to record checksums. So this lock has empty checksums.
    let input = include_str!("../tests/inputs/Gemfile.lock.withoutsource");
    let output = must_parse(input);
    insta::assert_yaml_snapshot!(output);
}

fn must_parse(input: &str) -> crate::datatypes::GemfileDotLock<'_> {
    match crate::parse(input) {
        Ok(o) => o,
        Err(e) => {
            let report = miette::Report::new(e);
            panic!("{report:?}")
        }
    }
}
