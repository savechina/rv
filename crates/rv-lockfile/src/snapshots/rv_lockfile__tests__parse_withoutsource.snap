---
source: crates/rv-lockfile/src/tests.rs
expression: output
---
gem:
  - remote: "http://rubygems.org"
    specs:
      - gem_version:
          name: rack
          version: 3.2.3
        deps: []
platforms:
  - aarch64-linux-gnu
  - aarch64-linux-musl
  - arm-linux-gnu
  - arm-linux-musl
  - arm64-darwin
  - x86_64-darwin
  - x86_64-linux-gnu
  - x86_64-linux-musl
dependencies:
  - name: rack
    semver: ~
    nonstandard: false
bundled_with: 2.7.2
checksums:
  - gem_version:
      name: rack
      version: 3.2.3
    algorithm: None
    value: []
